### PR TITLE
Add optional header flag for web requests

### PR DESCRIPTION
**Pull request by w4tson**
With help from @andyrbell I've added optional flag 'header' for adding http-headers to waits.

### CODE REVIEW

File: dockerize.go [hunk 1]
```diff
@@ -177,6 +192,21 @@ func main() {
...
 	}
+
+	for _, h := range headersFlag {
+		const errMsg = "bad HTTP Headers argument: %s. expected \"headerName:headerValue\""
+		if strings.Contains(h, ":") {
+			parts := strings.Split(h, ":")
+			if len(parts) != 2 {
+				log.Fatalf(errMsg, headersFlag)
+			}
+			headers = append(headers, HttpHeader{name: parts[0], value: parts[1]})
```
Comment (jwilder) [inline dockerize.go]: Should probably call `strings.Trim` on both parts.

File: dockerize.go [hunk 2]
```diff
@@ -155,6 +169,7 @@ func main() {
 	flag.Var(&stdoutTailFlag, "stdout", "Tails a file to stdout. Can be passed multiple times")
 	flag.Var(&stderrTailFlag, "stderr", "Tails a file to stderr. Can be passed multiple times")
 	flag.StringVar(&delimsFlag, "delims", "", `template tag delimiters. default "{{":"}}" `)
+	flag.Var(&headersFlag, "header", "HTTP headers, colon separated. e.g \"Accept-Encoding:gzip\". Can be passed multiple times")
```
Comment (jwilder) [inline dockerize.go]: `-header` might not be intuitive since this option is specifically for `-wait` and only applies to the `http` waits.  

Maybe something like `-wait-http-header` or `-wait-header` and the value as `http:"Accept-Encoding:gzip"` perhaps.  That might be easier to understand that it's specifically for wait and allow for the possibility headers for other protocols.

**Comment by jwilder** [comment]
Some questions:
* What happens if two `-wait http://...` options are passed?  
* What happens if `-headers` is passed, but no `-wait`?

**Comment by w4tson** [comment]
The renamed option will apply to all -wait options. If no -wait but -wait-http-header then it will fail sensibly.

File: README.md [hunk 3]
```diff
@@ -91,6 +91,12 @@ If your file uses `{{` and `}}` as part of it's syntax, you can change the templ
 $ dockerize -delims "<%:%>"
 ```
 
+Http headers can be specified for http/https protocols.
+
+```
+$ dockerize -wait http://web:80 -header "Authorization:Basic QWxhZGRpbjpvcGVuIHNlc2FtZQ=="
```
Comment (jwilder) [inline README.md]: `-wait-http-header`

File: dockerize.go [hunk 4]
```diff
@@ -71,12 +79,8 @@ func waitForDependencies() {
 	dependencyChan := make(chan struct{})
 
 	go func() {
-		for _, host := range waitFlag {
-			log.Println("Waiting for host:", host)
-			u, err := url.Parse(host)
```
Comment (jwilder) [inline dockerize.go]: Why was this removed?

File: dockerize.go [hunk 5]
```diff
@@ -177,6 +189,34 @@ func main() {
 			log.Fatalf("bad delimiters argument: %s. expected \"left:right\"", delimsFlag)
 		}
 	}
+
+	for _, host := range waitFlag {
+		u, err := url.Parse(host)
```
Comment (jwilder) [inline dockerize.go]: Ah, you moved it down here.

**Comment by jwilder** [comment]
One small issue w/ the README, but looks good.

**Comment by w4tson** [comment]
Cool. done

**Comment by jwilder** [comment]
@w4tson I don't see the updated `README.md` change w/ `-wait-http-header`?  Did you push it up?

**Comment by w4tson** [comment]
errrr, I have now.

**Comment by jwilder** [comment]
Thanks @w4tson!
